### PR TITLE
refactor: normalize terminology and project structure

### PR DESCRIPTION
**Pull request by LayZeeDK**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[x] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

1. Terminology is a little confusing and sometimes inconsistent.
1. Naming of files and folders is inconsistent.

Issue Number: N/A

## What is the new behavior?
1. Sticking to agreed terminology and public API export names.
1. Files and folders use a similar structure throughout the codebase

## Does this PR introduce a breaking change?

```
[x] Yes
[ ] No
```

**A LOT** of breaking changes before version 2.0.0.

## Other information
Terminology and structure review
- [x] `@ngworker/lumberjack`
- [x] `@ngworker/lumberjack/console-driver`
- [x] `@ngworker/lumberjack/http-driver`
- [x] READMEs
- [x] `@internal/test-util`
- [x] `@internal/test-util/console-driver`

### CODE REVIEW

File: libs/ngworker/lumberjack/http-driver/README.md [hunk 1]
```diff
@@ -1,115 +1,114 @@
...
 @NgModule()
-export class HttpDriverModule {
+export class LumberjackHttpDriverModule {
   /**
    * Pass a full HTTP driver configuration.
    */
-  static forRoot(config: HttpDriverConfig): ModuleWithProviders<HttpDriverRootModule> {
+  static forRoot(config: LumberjackHttpDriverConfig): ModuleWithProviders<LumberjackHttpDriverRootModule> {
     return {
       ngModule: HttpDriverRootModule,
```
Comment (NachoVazquez) [inline libs/ngworker/lumberjack/http-driver/README.md:113]:
```suggestion
      ngModule: LumberjackHttpDriverRootModule,
```

**Review by NachoVazquez — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -275,21 +275,21 @@ A driver module provides configuration and other dependencies to a log driver. I
...
-      provide: logDriverToken,
+      provide: lumberjackLogDriverToken,
       useClass: ConsoleDriver,
       multi: true,
     },
   ],
 })
 export class ConsoleDriverModule {
-  static forRoot(config?: LogDriverConfig): ModuleWithProviders<ConsoleDriverModule> {
+  static forRoot(config?: LumberjackLogDriverConfig): ModuleWithProviders<ConsoleDriverModule> {
```
[the remark this answers is not in the document]
Comment (LayZeeDK) [reply README.md:292]: No, this is a custom, simplified console log driver implementation for this guide.

File: README.md [hunk 3]
```diff
@@ -304,18 +304,18 @@ If no configuration is passed, then the root `LogDriverConfig` is used.
 
 ```ts
 import { InjectionToken } from '@angular/core';
-import { LogDriverConfig, logDriverConfigToken } from '@ngworker/lumberjack';
+import { LumberjackLogDriverConfig, lumberjackLogDriverConfigToken } from '@ngworker/lumberjack';
 
-export const consoleDriverConfigToken = new InjectionToken<LogDriverConfig>('Console driver configuration', {
-  factory: () => inject(logDriverConfigToken),
+export const consoleDriverConfigToken = new InjectionToken<LumberjackLogDriverConfig>('__CONSOLE_DRIVER__CONFIG__', {
```
[the remark this answers is not in the document]
Comment (LayZeeDK) [reply README.md:309]: >Should we add consistency to the tokens value like we do we services and modules names?

No, this is a custom, simplified console log driver implementation for this guide. I don't want readers to confuse it with the `LumberjackConsoleDriver` implementation.

File: libs/internal/test-util/src/index.ts [hunk 4]
```diff
@@ -1,15 +1,31 @@
...
- * Public API Surface of @internal/test-util
+ * Public API surface of @internal/test-util
  */
 
+// Angular
+export * from './lib/angular/expect-ng-module-to-be-guarded-against-direct-import';
+export * from './lib/angular/resolve-dependency';
+
+// Error-throwing driver
+export * from './lib/error-throwing-driver/error-throwing-driver.config';
```
[the remark this answers is not in the document]
Comment (LayZeeDK) [reply libs/internal/test-util/src/index.ts:10]: I don't think it's worth it for testing drivers. It's easier to differentiate testing ones from public ones just by looking at their names. It's also a lot less typing.

File: libs/ngworker/lumberjack/console-driver/src/configuration-api.spec.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import { isClass } from '@internal/test-util';
+
+import { LumberjackConsoleDriverModule } from './index';
+
+describe('Configuration API', () => {
+  describe('Angular modules', () => {
+    it(`exposes ${LumberjackConsoleDriverModule.name}`, () => {
+      const sut = LumberjackConsoleDriverModule;
```
Comment (NachoVazquez) [inline libs/ngworker/lumberjack/console-driver/src/configuration-api.spec.ts:8]: sut stands for what?

Comment (LayZeeDK) [reply libs/ngworker/lumberjack/console-driver/src/configuration-api.spec.ts:8]: Subject Under Test

**Review by NachoVazquez — APPROVED** [review]

**Pull request merged**